packages/adblocker/src/filters/network.ts

```diff
@@ -400,6 +400,20 @@ function compileRegex(
   return new RegExp(filter);
 }
 
+function findLastIndexOfUnescapedCharacter(text: string, character: string) {
+  let lastIndex = text.lastIndexOf(character);
+
+  if (lastIndex === -1) {
+    return -1;
+  }
+
+  while (text.charCodeAt(lastIndex - 1) === 92 /* '\\' */) {
```

Comment:
Nit. We need to check that « lastIndex » is not equal to 0 here.

Reply:
If the `lastIndex` is zero, it means there's no escape character before the character. We don't need to check if the variable is set to zero.

Reply:
Yes but your `chatCodeAt` will be out of bound in that case and that should be avoided IMO.

Reply:
It won't throw an exception because it'll return `NaN` in that case. For example:
```
Welcome to Node.js v20.11.0.
Type ".help" for more information.
> ''.charCodeAt(-1)
NaN
> .exit
```

But your way should be safer. I updated the code. Thanks.